spec/unit/provider/service/smf_spec.rb

```diff
@@ -174,17 +174,42 @@
   end
 
   describe "when restarting" do
-    it "should call 'svcadm restart /system/myservice'" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
-      @provider.expects(:wait).with('online')
-      @provider.restart
-    end
 
     it "should error if timeout occurs while restarting the service" do
-      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "/system/myservice"], true)
+      @provider.expects(:texecute).with(:restart, ["/usr/sbin/svcadm", :restart, "-s", "/system/myservice"], true)
       Timeout.expects(:timeout).with(60).raises(Timeout::Error)
       expect { @provider.restart }.to raise_error Puppet::Error, ('Timed out waiting for /system/myservice to transition states')
     end
 
+    context 'with :kernelrelease == 5.10' do
+      it "should call 'svcadm restart /system/myservice'" do
+        Facter.instance_variable_get(:@collection).instance_variable_get(:@facts).delete_if{|k,v| k==:kernelrelease}
+        Facter.stubs(:fact).with(:kernelrelease).returns Facter.add(:kernelrelease) { has_weight 500; setcode { '5.10' } }
```

Comment:
I've reduced the scope of the Facter change. I thought it should only be affecting the smaller set of tests but the timeout test still fails.